src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,113 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
+                        // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
+                        // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
+                        IE::Task t;
+                        do {
+                            _inferPipelineTasks.try_pop(t);
+                        } while (t && ScheduleToWorkerInferRequest(std::move(t)));
+                        do {
+                            _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
+                        } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    }
                 }
             });
     }
 }
 
+void AutoSchedule::releaseActualdevice(std::string currentDeviceName) {
+    _loadContext[ACTUALDEVICE].isEnabled = false;
+    _loadContext[ACTUALDEVICE].isLoadSuccess = false;
+    _loadContext[ACTUALDEVICE].isAlready = false;
+    _loadContext[ACTUALDEVICE].future.wait();
+    // clean up ACTUALDEVICE infer requests
+    size_t destroynum = 0;
+    std::pair<int, WorkerInferRequest*> worker;
+    while (_idleWorkerRequests[currentDeviceName].try_pop(worker)) {
+        destroynum++;
+    }
+    if (destroynum == _workerRequests[currentDeviceName].size()) {
+        std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+        _workerRequests[currentDeviceName].clear();
+        _loadContext[ACTUALDEVICE].executableNetwork._ptr.reset();
+        _loadContext[ACTUALDEVICE].executableNetwork._so.reset();
+    }
+    LOG_INFO_TAG("ACTUALDEVICE released!!");
+}
+
+bool AutoSchedule::selectOtherDevice(std::string currentDeviceName) {
+    {
+        std::lock_guard<std::mutex> lock(_autoSContext->_fallbackMutex);
+        if (_autoSContext->_modelPath.empty())
+            _loadContext[FALLBACKDEVICE].networkPrecision = GetNetworkPrecision(_autoSContext->_network);
+        _loadContext[FALLBACKDEVICE].metaDevices = _autoSContext->_devicePriorities;
+        if (currentDeviceName == "CPU_HELP") {
+            currentDeviceName = "CPU";
+        }
+        const auto CurrentDeviceIter = std::find_if(_autoSContext->_devicePriorities.begin(), _autoSContext->_devicePriorities.end(),
+                                            [=](const DeviceInformation& d) -> bool {
+                                            return d.deviceName.find(currentDeviceName) != std::string::npos;});
+        if (CurrentDeviceIter != _autoSContext->_devicePriorities.end()) {
+            if (_autoSContext->_devicePriorities.size() == 1) {
+                LOG_INFO_TAG("No other devices in _devicePriorities");
+                return false;
+            }
+            _autoSContext->_devicePriorities.erase(CurrentDeviceIter);
+        } else {
+            LOG_INFO_TAG("Alreay selected fallback device");
+            return _loadContext[FALLBACKDEVICE].isReloadSuccess ? true : false;
```

Comment:
at the time of return, the fallback device may not finish loading, this infer request will fail still

Reply:
probably need to check the load finished here: _cond.wait(lck, []{return loadfinished;});